projects/storefrontlib/src/cms-components/user/login/login.component.ts

```diff
@@ -1,33 +1,18 @@
 import { Component } from '@angular/core';
-import { AuthService, User, UserService } from '@spartacus/core';
+import { User, UserService } from '@spartacus/core';
 import { Observable } from 'rxjs';
-import { filter, map, switchMap } from 'rxjs/operators';
-import { LoginComponentService } from './login.component.service';
+import { map } from 'rxjs/operators';
 
 @Component({
   selector: 'cx-login',
   templateUrl: './login.component.html',
 })
 export class LoginComponent {
-  constructor(
-    private auth: AuthService,
-    private userService: UserService,
-    private loginService: LoginComponentService
-  ) {}
+  constructor(private userService: UserService) {}
 
   get user$(): Observable<User> {
-    return this.auth.getUserToken().pipe(
-      map(token => {
-        if (token && !!token.access_token && !this.loginService.isLogin) {
-          this.loginService.isLogin = true;
-          this.userService.load(token.userId);
-        } else if (token && !token.access_token && this.loginService.isLogin) {
-          this.loginService.isLogin = false;
-        }
-        return token;
-      }),
-      filter(token => token && !!token.access_token),
-      switchMap(() => this.userService.get())
-    );
+    return this.userService
+      .get()
+      .pipe(map(user => (Object.keys(user).length !== 0 ? user : undefined)));
```

Comment:
Just thought about this, the auth guard (and not-auth guard) validate check for the user token to see if the user is connected. Should they be changed to use this logic?